world/stat_checks/check_maker.py

```diff
@@ -88,6 +94,68 @@ def announce_to_room(self):
             self.roll_message, options={"roll": True}
         )
 
+    def announce_to_players(self):
+        """
+        Sends a private roll result message to specific players as well as
+        all staff at self.character's location.
+        """
+        self_list = (
+            "me",
+            "self",
+            str(self.character).lower(),
+            str(self.character.key).lower(),
+        )
+
+        # Adjust receiver list; this is who actually will receive it besides staff and the caller.
+        receiver_list = self.receivers.copy()
+        for name in self.receivers:
+            receiver = self.character.search(name.strip(), use_nicks=True)
```

Comment:
While we can do the search here, in general I'd prefer if we did the search in the command and passed the actual objects instead. Two reasons for that: I think from a separation of concerns standpoint the command should be responsible for interpreting player input, so anything related to parsing player input is better handled there, and also so that we can more easily handle cases like rolls where receivers are in different rooms. They might be specified by some other mechanism rather than just by name: for example, it could be everyone on an action together. When generating a roll for that, we'd want to pass along the receivers based on the action, rather than based on a command - making it based purely on names here and executing a search prevents that from working.